### PR TITLE
ported the evio-converter/decoder to the branch

### PR DESCRIPTION
**Pull request by gavalian**
introduced a new executable that allows converting EVIO files to EVIO6 format, this is convenient to have files larger than 2GB to be processed on the farm. Then the EVIO files can be converted to HIPO5, preserving all branches and from EVIO format. The HIPO5 then is converted to HIPO4 where the banks are created for each detector.

./coatjava/bin/recon-stream -convert -o clas_018775_0000_00001.ev6 class_018775_00000.evio class_018775_00001.evio
./coatjava/bin/recon-stream -decode -o clas_018775_0000_00001.ev6.h5 clas_018775_0000_00001.ev6 -m 3 

Will convert to HIPO5 with all the translation tables applied (use -m 2 to leave the data untranslated, i.e. create/slot/channel)

./coatjava/bin/recon-stream -decode -o clas_018775_0000_00001.ev6.h5 clas_018775_0000_00001.ev6 -m 4

The mode 4 will finish by converting HIPO5 to HIPO4, and can be used by the reconstruction engines.

### CODE REVIEW

**Comment by gavalian** [comment]
Switching to JDK 17 required the change of spot bugs version, (versions<4.0) do not work with JDK>14, switching to version 4.2.2 seemed to fix the problem, however, now the build fails with an error message about clas-utils.

Apparently, it finds the bug there, but can't figure out from the output what was failing.

[ERROR] Failed to execute goal com.github.spotbugs:spotbugs-maven-plugin:4.2.2:spotbugs (spotbugs) on project clas-utils: Execution spotbugs of goal com.github.spotbugs:spotbugs-maven-plugin:4.2.2:spotbugs failed: Java returned: 1 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.github.spotbugs:spotbugs-maven-plugin:4.2.2:spotbugs (spotbugs) on project clas-utils: Execution spotbugs of goal com.github.spotbugs:spotbugs-maven-plugin:4.2.2:spotbugs failed: Java returned: 1

**Comment by c-dilks** [comment]
Java version and distribution is also fixed to `11, zulu` in `clas12-validation`, so those pipelines will just fail for now. Ping me when you've got this PR working and I'll update the versions in `clas12-validation` too.

**Comment by c-dilks** [comment]
To test validation pipelines with Java 17, apply this patch here:
```diff
diff --git a/.github/workflows/validation.yml b/.github/workflows/validation.yml
index c79002359..6c264a795 100644
--- a/.github/workflows/validation.yml
+++ b/.github/workflows/validation.yml
@@ -9,4 +9,4 @@ on:

 jobs:
   validation:
-    uses: JeffersonLab/clas12-validation/.github/workflows/ci.yml@main
+    uses: JeffersonLab/clas12-validation/.github/workflows/ci.yml@java17
```
Revert the patch before marking this PR ready.

**Comment by c-dilks** [comment]
I pushed a small fix in `clas12-validation` and re-triggered its pipeline from here; looks like it passes now. I'll go ahead and merge that there.

However, it looks like the `kpp` test is failing local tests here (that one is not tested in `clas12-validation`): https://github.com/JeffersonLab/coatjava/actions/runs/7919196292/job/21619678027?pr=190#step:6:7261

Maybe something is still stuck on Java 11? It's odd that only one of the tests fails...

File: .github/workflows/validation.yml [hunk 1]
```diff
@@ -9,4 +9,4 @@ on:
 
 jobs:
   validation:
-    uses: JeffersonLab/clas12-validation/.github/workflows/ci.yml@main
+    uses: JeffersonLab/clas12-validation/.github/workflows/ci.yml@java17
```
Comment (c-dilks) [inline .github/workflows/validation.yml]: Revert this, now that https://github.com/JeffersonLab/clas12-validation/pull/87 is merged (and the `java17` branch is deleted)
```suggestion
    uses: JeffersonLab/clas12-validation/.github/workflows/ci.yml@main
```

Comment (c-dilks) [reply .github/workflows/validation.yml]: resolved by #191

**Comment by c-dilks** [comment]
> However, it looks like the `kpp` test is failing local tests here (that one is not tested in `clas12-validation`): https://github.com/JeffersonLab/coatjava/actions/runs/7919196292/job/21619678027?pr=190#step:6:7261
> 
> Maybe something is still stuck on Java 11? It's odd that only one of the tests fails...

resolved by #191

**Pull request closed without merging**